### ISSUE TITLE
Can we get ip geolocation back on UT?

### ISSUE THREAD

**Issue by mauricioduarte01** [ubports/ubuntu-touch#1188]
**Description of the feature**

The location of my device seems to work super-slowly when trying to find the initial fix. I have tested on Unav, Google maps with Morphbrowser, and Sensorsstatus. This started happening once we decided to remove wolfpack and move to raw gps.

    Device:Hammerhead_Nexus_5
    Channel: RC/Devel/Edge
    Build: OTA-10

**Steps to reproduce**

Make a localization request by an application or a web page. If you used to have a quick initial fix now will most likely be a problem, specially if you are trying inside a building or under a roof etc.

**Expected behavior**

Initial fix should be fast and reliable. (you can still check that this is the case when using the Stable channel where "wolfpack and ip geoclue" weren't removed yet)

The big question here is, can we get ip geolocation back?

Thank you in advance. Keep up the good work.

**Comment by Flohack74** [comment]
We are working on shifting to another backend API for location. But due to the troubles we had with the old one, there will be some time without assisted GPS and I can just ask for your patience.

**Comment by UniversalSuperBox** [comment]
While the geoclue location backend was good for some, it caused issues for others:

https://github.com/ubports/ubuntu-touch/issues/24 (closed but there were more reports of this up to a month ago)
https://github.com/ubports/ubuntu-touch/issues/1114
https://github.com/ubports/ubuntu-touch/issues/282
https://github.com/ubports/ubuntu-touch/issues/1130

Often the assist data was kilometers, even countries, off of the real location. The service itself didn't seem very stable either, taking the entire location service down if it experienced a problem. I approved the pull requests that removed geoclue (https://github.com/ubports/lxc-android-config/pull/25 and https://github.com/ubports/ubuntu-touch-meta/pull/48) because of these problems. Now that we're back down to a small base, we can build back up with better technology in a more stable way.

So, as Florian says, there will be some time without assisted GPS. However, we believe the trade-off of lost quick lock for more reliable location data is worth it. You can see https://docs.ubports.com/en/latest/userguide/dailyuse/location.html to learn about improving your GPS lock time. It's not perfect, but it should be reliable.

In short: Assisted GPS is coming back, but we don't expect to put back what was there before to do it.

**Comment by dobey** [comment]
Just to be clear. We still have AGPS (at least for qcom devices), based on cell tower signal. It works well on my Nexus 4, where without a SIM I could not get a GPS lock after ~40 min sitting in window sill with screen on. Adding a SIM (even unregistered, so it could ping the towers), with WiFi on, it got an accurate location within a few minutes.

What we removed is the very broken geoip only hack that had no real licensing, and which was extremely unreliable and crashy. If it actually worked for you, then you were quite lucky.

**Comment by mauricioduarte01** [comment]
Thank you guys a lot for the answers. I completely understand why this happened. I just wanted to report this so we can keep a record on this. I am sure this will change for better in the future, and maybe this also will help to call the attention of other people that may help contributing with finding the best solution for the issue.
